packages/mui-material/src/Link/Link.js

```diff
@@ -67,7 +67,7 @@ const LinkRoot = styled(Typography, {
     }),
     ...(ownerState.underline === 'always' && {
       textDecoration: 'underline',
-      textDecorationColor: color !== 'inherit' ? alpha(color, 0.4) : undefined,
+      textDecorationColor: color !== 'inherit' && color !== null ? alpha(color, 0.4) : undefined,
```

Comment:
I believe it is a safe change that we can make. Let's proceed with updating the migration guide and adding a test case.